crnk-core/src/main/java/io/crnk/core/engine/internal/jackson/LinksInformationSerializer.java

```diff
@@ -34,13 +38,15 @@ public void serialize(LinksInformation value, JsonGenerator gen, SerializerProvi
 
 		for (String attrName : info.getAttributeNames()) {
 			BeanAttributeInformation attribute = info.getAttribute(attrName);
```

Comment:
the setup can be slightly simplified: BeanAttributeInformation already have a getJsonName that can be used here. So the LinksInformationSerializer does not need the propertyNameResolver, but can work with those information objects.
It just has not been used so far.

But your introduced JacksonPropertyNameResolver still  is a good idea. Probably best to integrate it with BeanAttributeInformation/BeanInformation. There is one not so nice thing tough. BeanInformation is fully static so far. So if we start making it customizable, that customization has to be static as well. For now  that seems ok-ish. Maybe in the future we have a BeanInformationFactory that is available from CrnkBoot to avoid the static/global JVM-specific things. Maybe as a start we could create that BeanInformationFactory. There would be a singleton default instance BeanInformationFactory.DEFAULT or something like that. BeanInformatioan.of(...) would make use of that singleton. And the BeanInformationFActory would have a registerNameResolver method.